api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -238,3 +246,56 @@ func (r *OpenStackDataPlaneNodeSetSpec) duplicateNodeCheck(nodeSetList *OpenStac
 
 	return
 }
+
+// Compare TLS settings of control plane and data plane
+// if control plane name is specified attempt to retrieve it
+// otherwise get any control plane in the namespace
+func (r *OpenStackDataPlaneNodeSetSpec) ValidateTLS(namespace string) error {
+	var err error
+
+	controlPlane := openstackv1.OpenStackControlPlane{}
+	if len(r.ControlPlane) > 0 {
+		controlPlaneName := types.NamespacedName{
+			Name:      r.ControlPlane,
+			Namespace: namespace,
+		}
+		err = webhookClient.Get(context.TODO(), controlPlaneName, &controlPlane)
+		if err == nil {
+			err = r.TLSMatch(controlPlane)
+		}
+
+	} else {
+		controlPlanes := openstackv1.OpenStackControlPlaneList{}
+		opts := client.ListOptions{
+			Namespace: namespace,
+		}
+
+		_ = webhookClient.List(context.TODO(), &controlPlanes, &opts)
+		// Verify TLS status of control plane only if there is a single one
+		// report error if there are multiple, or proceed if there are none
+		if len(controlPlanes.Items) > 1 {
+			err = fmt.Errorf("multiple control planes found in the namespace %v", namespace)
+		} else if len(controlPlanes.Items) == 1 {
+			controlPlane = controlPlanes.Items[0]
+			err = r.TLSMatch(controlPlane)
+		}
```

Comment:
Again, if there is no control plane , we should requeue and wait, right?

Reply:
ditto